# R_tutorials
**Chen**

Some R tutorials: foundation and bioinformatics.

Although most code have been edited, the reference of the origin code were presented before the tables of contents, only for education purpose. Please purchase the books for detailed explaination. All data are in .csv format.

### Table of Contents: Foundation
Reference: *Getting Started with R: An Introduction for Biologists, 2ed (Oxford, 2017)*
| Chapter    | Title                                     |
| ---------- | ----------------------------------------- |
| **Foundation** | [Data management with dplyr](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Data%20management%20with%20dplyr.r)     |
|  | [Visualising data with ggplot2 I](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Visualising%20data%20with%20ggplot2%20I.r) |
|  | [Statistics I](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Statistics%20I.r)      |
|  | [Statistics II](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Statistics%20II.r)    |
|  | [Generalized linear models](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Generalized%20linear%20models.r) |
|  | [Visualising data with ggplot2 II](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Visualising%20data%20with%20ggplot2%20II.r) |
| **Reference**  | [Themes in ggplot2](https://github.com/liuchen37/R_tutorials/blob/main/Reference:%20Themes%20in%20ggplot2.r)     |
| **Practice**   | [Inhibitory effect of DEA and 6-MP on T47D](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20Inhibitory%20effect%20of%20DEA%20and%206-MP%20on%20T47D.r) |
|    | [HRas mutation rate](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20Counting%20freqency%20of%20an%20object%20in%20a%20table.r) |
| **Data**       | [Compensation](https://github.com/liuchen37/R_tutorials/blob/main/compensation.csv)     |
|        | [Breast Cancer](https://github.com/liuchen37/R_tutorials/blob/main/BC.csv)  |
|        | [Daphniagrowth](https://github.com/liuchen37/R_tutorials/blob/main/Daphniagrowth.csv)      |
|        | [GardenOzone](https://github.com/liuchen37/R_tutorials/blob/main/GardenOzone.csv)            |
|        | [Ladybirds_morph_colour](https://github.com/liuchen37/R_tutorials/blob/main/ladybirds_morph_colour.csv)    |
|        | [Plant.growth.rate](https://github.com/liuchen37/R_tutorials/blob/main/plant.growth.rate.csv)         |
|        | [Weight-height](https://github.com/liuchen37/R_tutorials/blob/main/weight-height.csv)                   |
|        | [Cow Growth](https://github.com/liuchen37/R_tutorials/blob/main/growth.csv)          |
|        | [Limpet](https://github.com/liuchen37/R_tutorials/blob/main/limpet.csv)    |

### Table of Contents: Bioinformatics
Reference: *R Bioinformatics Cookbook (Packt, 2019)*
| Chapter    | Title     |
| ---------- | ----------------------------------------- |
| **Quantitative RNAseq** | [Introduction and technical requirements](https://github.com/liuchen37/R_tutorials/blob/main/Quantitative%20RNAseq:%20Introduction%20and%20technical%20requirements.md) |
|  | [Estimating differential expression](https://github.com/liuchen37/R_tutorials/blob/main/Quantitative%20RNAseq:%20Estimating%20differential%20expression.r) |
|  | Power analysis |
|  | Unannotated transcribed regions |
|  | High expression region with bumphunter |
|  | Differential peak analysis |
|  | Estimating batch effects |
|  | Allele-specific expression analysis |
|  | RNAseq data visulaization |  |
| **Genetic variants and HTS data** | Introduction and technical requirements |
|  | SNPs and indels from sequencing data |
|  | Predicting ORFs in long reference sequences |
|  | Plotting with karyploteR |
|  | Variants selection and classifying |
|  | Genomic regions of interest |
|  | GWAS |
|  | Estomating the copy number at a locus of interest |
| **Genes, proteins, domains and motifs** | Introduction and technical requirements |
|  | DNA motif |
|  | Protein domains with PFAM and bio3d |
|  | InterPro domains |
|  | Multiple alignments of genes and proteins |
|  | Aligning genomic length sequences |
|  | Machine learning for protein noval feature predicton |
|  | 3D structure protein alignment |
| **Phylogenetics** | Introduction and technical requirements |
|  | Reading and writing varied tree formats |
|  | Tree visualization |
|  | Quantifying differences |
|  | Subtrees |
|  | Alighnment visualization |
|  | Reconstructing trees using phangorn |
| **Metagenomics** | Introduction and technical requirements |
|  | Processing hierarchical taxonomic data |
|  | Sample differential |
|  |Reading amplicon data |
|  |Taxonomic abundances visualization |
|  |Sample diversity |
|  |Splitting sequence files into OTUs |
| **Proteomics** | Introduction and technical requirements |
|  | Visualization raw MS data |
|  | Genome broswer |
|  | Distribution visualizing |
|  | MS formats convertion |
|  | Matching spectra to peptides |
|  | QC files |
|  |Identifying genomic loci with peptides |
| **Visualization for publication** | [Introduction and technical requirements](https://github.com/liuchen37/R_tutorials/blob/main/Visualization%20for%20publication%09Introduction%20and%20technical%20requirements.md) |
|  | [Mutiple distributions](https://github.com/liuchen37/R_tutorials/blob/main/Visualization%20for%20publication:%20Mutiple%20distributions.r) |
|  | [Colormpas for two-variable data](https://github.com/liuchen37/R_tutorials/blob/main/Visualization%20for%20publication:%20%09Colormpas%20for%20two-variable%20data.r) |
|  | Relational data as networks |
|  | Interactive web graphics |
|  | Three-dimensional plots |
|  | Circular genome plots |
| **Databases and remote data sources** | Introduction and technical requirements |
|  | Retrieving gene and genoe annotation |
|  | SNPs |
|  | Gene ontology |
|  | SRA/ENA |
|  | Performing QC and filtering on high-throughput sequence reads |
|  | Completing read-to-reference alignments |
|  | Visualizing QC of read-to-reference alignments |
| **Statistics and machine learning** | Introduction and technical requirements |
|  | Correcting p-values for multiple hypotheses |
|  | Simulated dataset and classifying |
|  | Learning groups I |
|  | Predicting classes I: random forests |
|  | Predicting classes II: SVM |
|  | Learning groups II |
|  | Identifying important variables in data I |
|  | Identifying important variables in data II |
| **Tidyverse and BioConductor** | Introduction and technical requirements |
|  | Tidying |
|  | Nested dataframes |
|  | dplyer::mutate() |
|  | BioConductor classes |
|  | Developing reusable workflows and reports |
|  | Family of functions |
| **Building and packaging** | Introduction and technical requirements |
|  | Simple S3 objects |
|  | S3 classes |
|  | S4 systems |
|  | Packaging for sharing and reuse |
|  | GitHub |
|  | Building a unit test suite |
|  | Continous integration |
| **Practice**   | [GEOquery](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20GEOquery.r) |


**Keep updating, keep learning...**
